### DISCUSSION TITLE
Which is faster C or V?

### DISCUSSION THREAD

**Discussion by MrMidnightOfficial** [vlang/v#14781] [Polls]
Which is faster C or V?

**Comment by JalonSolov** [comment]
They're about the same.  V converts to C, which is then compiled by a C compiler, so it will be "C speed".

The main difference is in what algorithm you use.

**Comment by JDoawp** [comment]
As mentioned, V converts to C. So just like C vs ASM it'll only ever be as fast as the other part.

Although just like C vs ASM you can probably write faster code in V unless you're good at C.

I think if you take the ease of writing, features and all that into account V is faster, at least for an average Java schmuck like me :)

**Reply by MrMidnightOfficial** [reply]
TYSM

**Comment by ghost** [comment]
Since V is currently a transpiler, if you write C code that is exactly the same as the C code generated by the V compiler, the speed is the same for both V and C.
The codes can be compared, but the languages themselves cannot.

**Comment by Wajinn** [comment]
I think this poll is a missing a 3rd option:

3) Is about the same speed or doesn't usually make a difference.

As mentioned by JDoawp and Kasha, V converts to C, so the difference in speed will often come down to the type of code and algorithm, or the difference is negligible enough not to matter.  

A lot of people get into this odd speed demon frenzy, but when referring to the compiled languages, they often are not doing something where the speed is critical or the differences are quite small.  If there is something where speed really matters, they can then optimize their V code as necessary, and you do have the ability to add assembly too.
